source/Session.cpp

```diff
@@ -121,6 +121,29 @@ bool Session::hasNextCondition() const{
 	return false;
 }
 
+const RealTime Session::targetFrameTime()
+{
+	if (m_config->render.frameTimeArray.size() == 0) {
+		// The below matches the functionality in FPSciApp::updateParameters()
+		if (m_config->render.frameRate > 0) {
+			return 1.0f / m_config->render.frameRate;
+		}
+		else {
+			return 1.0f / float(m_app->window()->settings().refreshRate);
+		}
+	}
+	else {
+		if (m_config->render.randomFrameTime) {
+			return m_config->render.frameTimeArray.randomElement();
+		}
+		else {
+			static uint targetIdx = 0;
+			return m_config->render.frameTimeArray[targetIdx++ % m_config->render.frameTimeArray.size()];
```

Comment:
I think the indexing here might introduce a minor bug/benefit from some simplification.

Since `targetIdx` is a static uint that is never reset it will eventually wrap around the max uint value. This will produce a discontinuity in the frame times. Though extremely uncommon, I think this can be solved w/ the following code:

```
static uint targetIdx = 0;
targetIdx += 1;
targetIdx %= m_config->render.frameTimeArray.size();
return m_config->render.frameTimeArray[targetIdx];
```

This prevents the overflow "discontinuity" event by  capping `targetIdx` at the frame time array size. I guess technically we should also document that there is a 2^32-1 limit on the length of the frame array, but even at 1kHz frame rate this is still something like 50 days of consecutive values... so likely not an issue.